flink-core/src/test/java/org/apache/flink/api/java/typeutils/FieldAccessorTest.java

```diff
@@ -0,0 +1,280 @@
+		assertEquals("aa", f0.get(t));
+		assertEquals("aa", t.f0);
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	public void tupleFieldAccessorOutOfBoundsTest() {
+		try {
+			FieldAccessor<Tuple2<Integer, Integer>, Integer> fieldAccessor =
+					(FieldAccessor<Tuple2<Integer, Integer>, Integer>) (Object)
+							TupleTypeInfo.getBasicTupleTypeInfo(Integer.class, Integer.class)
+									.getFieldAccessor(2, null);
+			fail();
+		} catch (IndexOutOfBoundsException e) {
+			// Nothing to do here
+		}
+	}
+
+	public static class Foo {
+		public int x;
+		public Tuple2<String, Long> t;
+		public Short y;
+
+		public Foo() {}
+
+		public Foo(int x, Tuple2<String, Long> t, Short y) {
+			this.x = x;
+			this.t = t;
+			this.y = y;
+		}
+	}
+
+	@Test
+	public void testTupleInPojoInTuple() {
+		Tuple2<String, Foo> t = Tuple2.of("aa", new Foo(8, Tuple2.of("ddd", 9L), (short) 2));
+		TupleTypeInfo<Tuple2<String, Foo>> tpeInfo =
+				(TupleTypeInfo<Tuple2<String, Foo>>)TypeExtractor.getForObject(t);
+
+		FieldAccessor<Tuple2<String, Foo>, Long> f1tf1 = tpeInfo.getFieldAccessor("f1.t.f1", null);
+		assertEquals(9L, (long) f1tf1.get(t));
+		assertEquals(9L, (long) t.f1.t.f1);
+		t = f1tf1.set(t, 12L);
+		assertEquals(12L, (long) f1tf1.get(t));
+		assertEquals(12L, (long) t.f1.t.f1);
+
+		FieldAccessor<Tuple2<String, Foo>, String> f1tf0 = tpeInfo.getFieldAccessor("f1.t.f0", null);
+		assertEquals("ddd", f1tf0.get(t));
+		assertEquals("ddd", t.f1.t.f0);
+		t = f1tf0.set(t, "alma");
+		assertEquals("alma", f1tf0.get(t));
+		assertEquals("alma", t.f1.t.f0);
+
+		FieldAccessor<Tuple2<String, Foo>, Foo> f1 = tpeInfo.getFieldAccessor("f1", null);
+		FieldAccessor<Tuple2<String, Foo>, Foo> f1n = tpeInfo.getFieldAccessor(1, null);
+		assertEquals(Tuple2.of("alma", 12L), f1.get(t).t);
+		assertEquals(Tuple2.of("alma", 12L), f1n.get(t).t);
+		assertEquals(Tuple2.of("alma", 12L), t.f1.t);
+		Foo newFoo = new Foo(8, Tuple2.of("ddd", 9L), (short) 2);
+		f1.set(t, newFoo);
+		assertEquals(newFoo, f1.get(t));
+		assertEquals(newFoo, f1n.get(t));
+		assertEquals(newFoo, t.f1);
+	}
+
+
+	public static class Inner {
+		public long x;
+		public boolean b;
+
+		public Inner(){}
+
+		public Inner(long x) {
+			this.x = x;
+		}
+
+		public Inner(long x, boolean b) {
+			this.x = x;
+			this.b = b;
+		}
+
+		@Override
+		public String toString() {
+			return ((Long)x).toString() + ", " + b;
+		}
+	}
+
+	public static class Outer {
+		public int a;
+		public Inner i;
+		public short b;
+
+		public Outer(){}
+
+		public Outer(int a, Inner i, short b) {
+			this.a = a;
+			this.i = i;
+			this.b = b;
+		}
+
+		@Override
+		public String toString() {
+			return a+", "+i.toString()+", "+b;
+		}
+	}
+
+	@Test
+	public void testPojoInPojo() {
+		Outer o = new Outer(10, new Inner(4L), (short)12);
+		PojoTypeInfo<Outer> tpeInfo = (PojoTypeInfo<Outer>)TypeInformation.of(Outer.class);
+
+		FieldAccessor<Outer, Long> fix = tpeInfo.getFieldAccessor("i.x", null);
```

Comment:
Unfortunately, we do need it. `ProductFieldAccessor` uses a serializer for copying (which is needed because of the immutability), and `createSerializer` expects the `ExecutionConfig`.